modules/tensor_mechanics/src/materials/ComputeLinearViscoelasticCreepStrain.C

```diff
@@ -0,0 +1,110 @@
+    _has_driving_eigenstrain(isParamValid("driving_eigenstrain")),
+    _driving_eigenstrain_prefactor(getParam<Real>("driving_eigenstrain_prefactor")),
+    _driving_eigenstrain_name(
+        _has_driving_eigenstrain ? getParam<std::string>("driving_eigenstrain") : ""),
+    _driving_eigenstrain(_has_driving_eigenstrain
+                             ? &getMaterialPropertyByName<RankTwoTensor>(_driving_eigenstrain_name)
+                             : NULL)
+{
+}
+
+void
+ComputeLinearViscoelasticCreepStrain::updateQpViscousStrain(unsigned int qp,
+                                                            const RankTwoTensor & strain,
+                                                            const RankTwoTensor & stress)
+{
+  if (_has_driving_eigenstrain)
+    _viscoelastic_model->updateQpViscousStrain(qp,
+                                               _viscous_strains[qp],
+                                               _viscous_strains_old[qp],
+                                               strain,
+                                               stress,
+                                               true,
+                                               (*_driving_eigenstrain)[qp] *
+                                                   _driving_eigenstrain_prefactor);
+  else
+    _viscoelastic_model->updateQpViscousStrain(
+        qp, _viscous_strains[qp], _viscous_strains_old[qp], strain, stress, false);
+}
+
+void
+ComputeLinearViscoelasticCreepStrain::initQpStatefulProperties()
+{
+  ComputeCreepStrainBase::initQpStatefulProperties();
+
+  unsigned int components = _viscoelastic_model->components(_qp);
+
+  _viscous_strains[_qp].resize(components, RankTwoTensor());
+  _viscous_strains_old[_qp].resize(components, RankTwoTensor());
+}
+
+void
+ComputeLinearViscoelasticCreepStrain::computeQpCreepStrain()
+{
+  if (_has_driving_eigenstrain)
+    _viscoelastic_model->accumulateQpViscousStrain(_qp,
+                                                   _creep_strain[_qp],
+                                                   _viscous_strains_old[_qp],
+                                                   true,
+                                                   (*_driving_eigenstrain)[_qp] *
+                                                       _driving_eigenstrain_prefactor);
+  else
+    _viscoelastic_model->accumulateQpViscousStrain(
+        _qp, _creep_strain[_qp], _viscous_strains_old[_qp], false);
+}
+
+MooseSharedPointer<LinearViscoelasticityBase>
+ComputeLinearViscoelasticCreepStrain::getViscoelasticModel(
+    const std::string & viscoelastic_model_name) const
+{
+  MooseSharedPointer<Material> test =
+      _mi_feproblem.getMaterial(viscoelastic_model_name, _material_data_type, _mi_tid, true);
+
+  if (!test)
+    mooseError(viscoelastic_model_name + " does not exist");
+
+  MooseSharedPointer<LinearViscoelasticityBase> viscoelastic_model =
+      MooseSharedNamespace::dynamic_pointer_cast<LinearViscoelasticityBase>(test);
```

Comment:
I am not sure. We need to get materials deriving from LinearViscoelasticityBase. I am not that familiar with Moose framework, so I don't really know how it could be possible without this pointer casting...